hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/infrastructure/graphite/CachingGraphiteClientTest.groovy

```diff
@@ -0,0 +1,50 @@
+package pl.allegro.tech.hermes.management.infrastructure.graphite
+
+import pl.allegro.tech.hermes.test.helper.cache.FakeTicker
+import spock.lang.Specification
+import spock.lang.Subject
+
+import java.time.Duration
+
+class CachingGraphiteClientTest extends Specification {
+    static final CACHE_TTL_IN_SECONDS = 30
+    static final CACHE_SIZE = 100_000
+
+    def underlyingClient = Mock(GraphiteClient)
+    def ticker = new FakeTicker()
+
+    @Subject
+    def cachingClient = new CachingGraphiteClient(underlyingClient, ticker, CACHE_TTL_IN_SECONDS, CACHE_SIZE)
+
+    def "should return metrics from the underlying client"() {
+        given:
+        underlyingClient.readMetrics("metric_1", "metric_2") >> new GraphiteMetrics([metric_1: "1", metric_2: "2"])
+
+        when:
+        def metrics = cachingClient.readMetrics("metric_1", "metric_2")
+
+        then:
+        metrics.metricValue("metric_1") == "1"
+        metrics.metricValue("metric_2") == "2"
+    }
+
+    def "should return metrics from cache while TTL does not expire"() {
+        when:
+        cachingClient.readMetrics("metric_1", "metric_2")
+        ticker.advance(Duration.ofSeconds(29))
+        cachingClient.readMetrics("metric_1", "metric_2")
+
+        then:
+        1 * underlyingClient.readMetrics("metric_1", "metric_2") >> new GraphiteMetrics([metric_1: "1", metric_2: "2"])
+    }
+
+    def "should get metrics from the underlying client after TTL expires"() {
+        when:
+        cachingClient.readMetrics("metric_1", "metric_2")
+        ticker.advance(Duration.ofSeconds(31))
```

Comment:
nit-picking: you could refer to `CACHE_TTL_IN_SECONDS` here, like `Duration.ofSeconds(CACHE_TTL_IN_SECONDS + 1)`.

Or replace `CACHE_TTL_IN_SECONDS` constant with sth like `CACHE_TTL` of type `Duration`, making this call like this: `CACHE_TTL.plusSeconds(1)`.

Similarly in the test before.